pkg/eks/components/NodeGroup.vue

```diff
@@ -0,0 +1,675 @@
+<script lang="ts">
+import Vue from 'vue';
+import { _EDIT } from '@shell/config/query-params';
+import LabeledSelect from '@shell/components/form/LabeledSelect.vue';
+import LabeledInput from '@components/Form/LabeledInput/LabeledInput.vue';
+import Checkbox from '@components/Form/Checkbox/Checkbox.vue';
+import KeyValue from '@shell/components/form/KeyValue.vue';
+import Banner from '@components/Banner/Banner.vue';
+import UnitInput from '@shell/components/form/UnitInput.vue';
+import { mapGetters, Store } from 'vuex';
+
+import { MANAGED_TEMPLATE_PREFIX, parseTags } from '../util/aws';
+
+import { isEmpty } from '@shell/utils/object';
+import debounce from 'lodash/debounce';
+import { randomStr } from '@shell/utils/string';
+
+// map between fields in rancher eksConfig and amazon launch templates
+const launchTemplateFieldMapping = {
+  imageId:      'ImageId',
+  userData:     'UserData',
+  instanceType: 'InstanceType',
+  ec2SshKey:    '',
+  resourceTags: 'TagSpecifications',
+  diskSize:     'BlockDeviceMappings'
+} as {[key: string]: string};
```

Comment:
Mentioned above, about other places as well, but can avoid the `as` here by just typing the const

const launchTemplateFieldMapping: {[key: string]: string}